File: my-folder/0907-koko-eating-bananas/solution.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int max=0;
        int low=1;
        for(int i=0;i<piles.length;i++){
            max=Math.max(max,piles[i]);
        }
        while(low<max){
            int m=(low+max)/2;
            int k=0;
            for(int val:piles){
                k+=Math.ceil((double)val/m);
            }
            if(k<=h){
                max=m;
            }
            else{
                low=m+1;
            }
        }
        return max;
    }   
}

File: my-folder/0907-koko-eating-bananas/solution.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int left=1;
        //int right=*max_element(piles.begin(),piles.end());
        int right=1e9;
        while(left<right){
            int mid=(left+right)/2;
            int totalhrs=0;
            for(auto pile:piles){
                totalhrs += (pile/mid)+(pile % mid != 0);
                if(totalhrs>h){
                    break;
                }
            }
            if(totalhrs<=h){
                right = mid;
            }
            else{
                left=mid+1;
            }

        }
        return left;
        
    }
};
